src/win32/fs.cpp

```diff
@@ -1085,22 +1060,35 @@ bool WinFileSystemAccess::chdirlocal(LocalPath& namePath) const
 #ifdef WINDOWS_PHONE
     return false;
 #else
-    string* name = namePath.editStringDirect();
-    name->append("", 1);
-    int r = SetCurrentDirectoryW((LPCWSTR)name->data());
-    name->resize(name->size() - 1);
-
+    int r = SetCurrentDirectoryW(namePath.localpath.c_str());
     return r;
 #endif
 }
 
-size_t WinFileSystemAccess::lastpartlocal(const string* name) const
+#if defined(_WIN32)
+size_t WinFileSystemAccess::lastpartlocal(const std::wstring* name) const
+{
+    for (size_t i = name->size(); i--;)
+    {
+        if ((name->data())[i] == '\\'
+            || (name->data())[i] == '/'
+            || (name->data())[i] == ':')
+        {
+            return (i + 1);
+        }
+    }
+
+    return 0;
```

Comment:
```suggestion
    auto p = name->find_last_of(L"\\/:");
    return p == string::npos ? 0 : p + 1;
```

Reply:
While it may be the right suggestion, the focus was not intended to change the original functions.